client/src/components/practice/long/Typing/index.tsx

```diff
@@ -0,0 +1,303 @@
+import { Box, Flex, Text, Textarea } from '@chakra-ui/react';
+import { disassemble } from 'hangul-js';
+import { useRouter } from 'next/router';
+import { useEffect, useRef, useState } from 'react';
+
+import DownArrow from '@/icons/DownArrow';
+import type { CharInfo } from '@/types/typing';
+import { getCharType } from '@/utils/char';
+import { getTypingAccuracy, getTypingSpeed, getTypingWpm, getWrongKeys } from '@/utils/typing';
+
+import useStopwatch from '../../short/useStopWatch';
+import TypingLine from '../common/TypingLine';
+import PracticeLongLayout from '../Layout';
+import InfoBar from './InfoBar';
+
+const INIT_INFO = {
+  wpm: 0,
+  accuracy: 0,
+  typist: 0,
+};
+
+interface PracticeLongTypingProps {
+  title: string;
+  content: string;
+  currPage: number;
+  totalPage: number;
+}
+
+export default function PracticeLongTyping({ content }: PracticeLongTypingProps) {
+  const router = useRouter();
+
+  // 원본 긴 글을 분석하여 저장
+  const contentInfos = useRef<CharInfo[]>(
+    [...content].map((char) => ({
+      char,
+      type: getCharType(char),
+      components: disassemble(char),
+    })),
+  );
+  // 사용자가 타이핑한 문자들을 분석하여 저장, 초기에는 빈 문자로 초기화
+  const typingInfos = useRef<CharInfo[]>(
+    [...content].map(() => ({
+      char: '',
+      type: 'other',
+      components: [],
+    })),
+  );
+  const [textarea, setTextarea] = useState('');
+  // 처음 'f'로 초기화되어 가장 먼저 온 글자에 포커싱, 'f' 이외에도 'c'(correct), 'i'(incorrect), 'u'(unknown)로 상태 구분
+  const [typingStates, setTypingStates] = useState<string>('f');
```

Comment:
```suggestion
  const [typingStates, setTypingStates] = useState('f');
```
이것과 이 밑에 useState에 타입을 빼줘도 될 것 같아요.